src/bsp/src/posix/src/UserInterface.cpp

```diff
@@ -17,6 +18,9 @@ int UserInterface::print(const char* format, va_list args) const {
 
     int retValue = vsnprintf(buffer, bufferSize, format, args);
 
+    vPortEnterCritical();
     ROS_INFO("%s", buffer);
+    vPortExitCritical();
```

Comment:
Was this for debug or ROS_INFO actually needs to be critical?

Reply:
Making it critical prevents a context switch (by disabling signals) in the middle of the function. I'd rather leave it there and be safe. FreeRTOS documentation does give printf as an example of a function that should be surrounded with disabled signals